crates/settings/src/settings_content.rs

```diff
@@ -234,6 +234,21 @@ pub enum BaseKeymapContent {
     None,
 }
 
+/// Position of window control buttons on Linux.
+///
+/// Valid values: "left" or "right"
+/// - "left": Window controls on the left side (macOS style)
+/// - "right": Window controls on the right side (Windows style)
+#[derive(Copy, Clone, Debug, Serialize, Deserialize, JsonSchema, MergeFrom, PartialEq, Eq, Default)]
+#[serde(rename_all = "snake_case")]
+pub enum WindowControlsPosition {
+    /// Window controls on the left side (macOS style)
+    Left,
+    /// Window controls on the right side (Windows style)
+    #[default]
+    Right,
+}
+
```

Comment:
Instead of making the user manually configure this in Zed's settings.json, this could be handled automatically by querying GSettings with the [gio crate](https://gtk-rs.org/gtk-rs-core/stable/latest/docs/gio/struct.Settings.html). The setting only impacts GNOME (even with GTK + libadwaita applications), so when Zed is not running on GNOME, the window controls should always be drawn on the right.

I can make a pull request for your branch implementing this.